app/Admin/Controllers/AgroDealerController.php

```diff
@@ -0,0 +1,339 @@
+        {
+            // Remove the default id filter
+            $filter->disableIdFilter();
+            $filter->like('first_name', __('admin.form.First name'))->select(\App\Models\AgroDealers::pluck('first_name', 'first_name'));;
+        });
+
+        //disable create button 
+        if ($user->inRoles(['agro-dealer'])) 
+        {
+            $grid->disableCreateButton();
+        }
+      
+    
+        $grid->column('agro_dealer_reg_number', __('admin.form.Agro-dealer registration number'))->display(function ($agro_dealer_reg_number) {
+            return $agro_dealer_reg_number ?? 'Not yet assigned';
+        })->sortable();
+        $grid->column('first_name', __('admin.form.First name'));
+        $grid->column('last_name', __('admin.form.Last name'));
+        $grid->column('email', __('admin.form.Email'));
+        $grid->column('category', __('admin.form.Agro Dealer Category'));
+        $grid->column('physical_address', __('admin.form.Physical address'));
+        $grid->column('region', __('admin.form.Region'));
+        $grid->column('status', __('admin.form.Status'))->display(function ($status) {
+            return Utils::tell_status($status)??'-';
+        })->sortable();
+
+      
+        $grid->column('id', __('admin.form.Certificate'))->display(function ($id) {
+            $agro_dealer =  AgroDealers::find($id);
+        
+            if ( $agro_dealer &&  $agro_dealer->status == 'accepted') {
+                $link = url('agro_certificate?id=' . $id);
+                return '<b><a target="_blank" href="' . $link . '">Imprimer le certificat</a></b>';
+            } else {
+            
+                return '<b>Aucun certificat disponible</b>';
+            }
+        });
+        
+        return $grid;
+    }
+
+    /**
+     * Make a show builder.
+     *
+     * @param mixed $id
+     * @return Show
+     */
+    protected function detail($id)
+    {
+        $show = new Show(AgroDealers::findOrFail($id));
+
+         //delete notification after viewing the form
+         Utils::delete_notification('AgroDealers', $id);
+
+         //check if the user is the owner of the form
+         $showable = Validation::checkUser('AgroDealers', $id);
+         if (!$showable) 
+         {
+             return('<p class="alert alert-danger">You do not have rights to view this form. <a href="/agro-dealers"> Go Back </a></p> ');
+         }
+       
+        $show->field('agro_dealer_reg_number', __('admin.form.Agro-dealer registration number'))->as(function ($value) {
+            return $value ?? 'Not yet assigned';
+        });
+        $show->field('first_name', __('admin.form.First name'));
+        $show->field('last_name', __('admin.form.Last name'));
+        $show->field('email', __('admin.form.Email'));
+        $show->field('category', __('admin.form.Agro Dealer Category'));
+        $show->field('physical_address', __('admin.form.Physical address'));
+        $show->field('region', __('admin.form.region'));
```

Comment:
this is admin.form.region but others are Region with a capital R that means one of them wont be able to be translated because the translation is case sensitive